oci/oci.go

```diff
@@ -116,6 +129,23 @@ func (r *Runtime) Name() string {
 // Depending if the container is privileged and/or trusted,
 // this will return either the trusted or untrusted runtime path.
 func (r *Runtime) Path(c *Container) string {
+	if c.runtimeHandler != "" {
+		runtimeHandler, ok := r.runtimes[c.runtimeHandler]
+		if !ok {
+			if c.runtimeHandler == UntrustedRuntime && r.untrustedPath != "" {
+				return r.untrustedPath
+			}
+
+			return r.trustedPath
+		}
+
+		if runtimeHandler.RuntimePath == "" {
+			return r.trustedPath
+		}
+
+		return runtimeHandler.RuntimePath
+	}
+
 	if !c.trusted {
```

Comment:
is everything below in line with the new additions above? I can still see untrustdPath references but I guess any inconsistencies would have had been caught in `validateConfig` right?

Reply:
I have add the code above so that in case of a `runtime_handler` provided by CRI, this will supersede the previous logic that is now below.
But I wanted to keep the same logic below in case we don't get any runtime handler, this way we keep the support of the deprecated options.

Also, once we completely remove untrusted runtime and the trust level (by 1.14 as agreed), this code will need to be revisited with some simpler names like `defaultRuntimePath` instead of `trustedPath`, and also we'll simplify the way the runtime handler is processed, since there won't be any fallback.

Reply:
totally agree